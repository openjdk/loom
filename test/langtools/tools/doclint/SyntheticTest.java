/*
 * Copyright (c) 2012, 2021, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/*
 * @test
 * @bug 8004832 8249634
 * @summary Add new doclint package
 * @modules jdk.javadoc/jdk.javadoc.internal.doclint
 * @build DocLintTester
 * @run main DocLintTester -Xmsgs:all -ref SyntheticTest.out SyntheticTest.java
 */

/**
 * This is a test that messages are not generated for synthesized elements
 * such as enum methods, and that a message *is* generated for a default constructor
 */
public class SyntheticTest {
    // No explicit constructor implies a default constructor

    /** enum E */
    public enum E {
        /** enum member E1 */
        E1,
        /** enum member E2 */
        E2,
        /** enum member E3 */
        E3
    }
}


