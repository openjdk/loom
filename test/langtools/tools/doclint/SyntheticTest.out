SyntheticTest.java:37: warning: use of default constructor, which does not provide a comment
public class SyntheticTest {
       ^
1 warning

